src/main/kotlin/com/pubnub/api/models/consumer/access_manager/v3/Grants.kt

```diff
@@ -37,7 +40,19 @@ internal data class PNChannelResourceGrant(
     override val get: Boolean = false,
     override val join: Boolean = false,
     override val update: Boolean = false
-) : PNResourceGrant(), ChannelGrant
+) : PNResourceGrant(), ChannelGrant, SpaceIdGrant {
```

Comment:
So users will be able to pass `PNChannelResourceGrant` to new `GrantToken`? Is this what we actually want?

Reply:
val grantTokenEndpoint = pubNubUnderTest.grantToken(
            ttl = expectedTTL,
            authorizedUserId = expectedAuthorizedUserId,
            spaceIds = listOf(
                SpaceIdGrant.name(spaceId = SpaceId(expectedSpaceIdValue), read = true, delete = true),
                SpaceIdGrant.pattern(pattern = expectedSpaceIdPattern, write = true, manage = true)
            ),
            userIds = listOf(
                UserIdGrant.id(userId = UserId(expectedUserIdValue), delete = true),
                UserIdGrant.pattern(pattern = expectedUserIdPattern, update = true)
            )
        )

PNChannelResourceGrant is internal  so it seems that it will be difficult for client to use `PNChannelResourceGrant` to new `GrantToken`

Other option would be to introduce new interfaces/classes that will be copy of existing interfaces/classes from Grants.kt. I can do this. Having them there will less work in the future when Channel/UUID related elements will be removed.